lib/vdsm/storage/lvm.py

```diff
@@ -657,10 +660,20 @@ def _reload_single_vg(self, vg_name):
             if error:
                 self._update_stale_vgs([vg_name])
 
+                if not out:
+                    # Reload a specific VG name and failing without
+                    # an output might be indicative of a real error.
+                    raise se.VolumeGroupDoesNotExist(vg_name, error=error)
+
                 # NOTE: vgs may return useful output even on failure, so we
                 # don't return here.
 
-            return self._updatevgs(out, [vg_name])
+            updated_vgs = self._updatevgs(out, [vg_name])
+
+            if vg_name not in updated_vgs:
+                raise se.VolumeGroupDoesNotExist(vg_name=vg_name)
```

Comment:
I don't think this is possible - if the command succeeded, we got output for the vg. If the output is invalid, we will raise InvalidOutputLine. Otherwise we will get a vg object. However if my assumption is wrong, raising is the best thing we can do. Maybe a comment that this should never happen?